test/OpenSpaceToolkit/Astrodynamics/Solver/LeastSquaresSolver.test.cpp

```diff
@@ -0,0 +1,422 @@
+
+        for (const auto& instant : instants)
+        {
+            const Real time = (instant - state.getInstant()).inSeconds();
+            const Real position = initialPosition * std::cos(time) + initialVelocity * std::sin(time);
+            const Real velocity = -initialPosition * std::sin(time) + initialVelocity * std::cos(time);
+
+            const Position newPosition = Position::Meters({position, 0.0, 0.0}, Frame::GCRF());
+            const Velocity newVelocity = Velocity::MetersPerSecond({velocity, 0.0, 0.0}, Frame::GCRF());
+
+            states.add(State(instant, newPosition, newVelocity));
+        }
+
+        return states;
+    };
+};
+
+class OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver_Step : public ::testing::Test
+{
+   protected:
+    const VectorXd xHat_ = VectorXd::Ones(6);
+    const Real rmsError_ = 1.0;
+    const LeastSquaresSolver::Step step_ = LeastSquaresSolver::Step(rmsError_, xHat_);
+};
+
+class OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver_Analysis : public ::testing::Test
+{
+   protected:
+    const Size observationCount_ = 10;
+    const String terminationCriteria = "Test Criteria";
+    const State solutionState_ = State(
+        Instant::J2000(),
+        Position::Meters({7.0e6, 0.0, 0.0}, Frame::GCRF()),
+        Velocity::MetersPerSecond({7.5e3, 0.0, 0.0}, Frame::GCRF())
+    );
+    const MatrixXd solutionCovariance_ = MatrixXd::Identity(6, 6);
+    const MatrixXd solutionFrisbeeCovariance_ = MatrixXd::Identity(6, 6);
+    const MatrixXd solutionResidualMatrix_ = MatrixXd::Ones(6, 6);
+    const Array<LeastSquaresSolver::Step> steps_ = {
+        LeastSquaresSolver::Step(2.0, VectorXd::Ones(6)), LeastSquaresSolver::Step(1.0, VectorXd::Ones(6))
+    };
+
+    const LeastSquaresSolver::Analysis analysis_ = LeastSquaresSolver::Analysis(
+        observationCount_,
+        terminationCriteria,
+        solutionState_,
+        solutionCovariance_,
+        solutionFrisbeeCovariance_,
+        solutionResidualMatrix_,
+        steps_
+    );
+};
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver_Step, Constructor)
+{
+    {
+        EXPECT_NO_THROW(LeastSquaresSolver::Step(rmsError_, xHat_));
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver_Step, StreamOperator)
+{
+    {
+        testing::internal::CaptureStdout();
+
+        EXPECT_NO_THROW(std::cout << step_ << std::endl);
+
+        const std::string output = testing::internal::GetCapturedStdout();
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver_Step, Print)
+{
+    {
+        testing::internal::CaptureStdout();
+
+        step_.print(std::cout);
+
+        const std::string output = testing::internal::GetCapturedStdout();
+    }
+}
+
+TEST_F(OpenSpaceToolkit_Astrodynamics_Solver_LeastSquaresSolver_Analysis, Constructor)
+{
+    {
+        EXPECT_NO_THROW(
+            LeastSquaresSolver::Analysis(
```

Comment:
Worth checking that the RMS error value is what you expect it to be since it is calculated in the constructor